module/zfs/dmu.c

```diff
@@ -860,6 +840,26 @@ dmu_free_long_range_impl(objset_t *os, dnode_t *dn, uint64_t offset,
 			return (err);
 		}
 
+		txg = dmu_tx_get_txg(tx);
+
+		mutex_enter(&dp->dp_lock);
+		long_free_dirty +=
+		    dp->dp_long_free_dirty_pertxg[txg & TXG_MASK];
```

Comment:
if I'm reading this correctly, we only increment `long_free_dirty` once, so it would be more clear to do:
```
uint64_t long_free_dirty = dp->dp_long_free_dirty_pertxg[txg & TXG_MASK];
```